### ISSUE TITLE
Problem regarding the OpenStack cloud-config

### ISSUE THREAD

**Issue by dabeck** [rancher/rancher#14197]
**Rancher versions:**
rancher/rancher:v2.0.3

**Type/provider of hosts:**
custom OpenStack provider

**Steps to Reproduce:**
Create a new Cluster within rancher 2.0.3 with a custom cloud-config and a OpenStack cloud-provider config:

```
...
cloud_provider:
  name: openstack
  openstackCloudProvider:
    global:
      auth-url: "https://example.../v3"
      domain-name: "domain"
      password: "password"
      region: "eu-de"
      tenant-id:  "id"
      username: "user"    
    load_balancer:
      subnet-id: "4ba20e2c-bb89-468f-81c6-2c67a4bac419"
      floating-network-id: "0a2228f2-7f8a-45f1-8e09-9039e1d09975"
```

**Results:**
The generated cloud-config is corrupted:
```
$ cat /etc/kubernetes/cloud-config 
[Global,omitempty]
auth-url,omitempty = https://example.../v3
username,omitempty = user
user-id,omitempty = 
password,omitempty = password
tenant-id,omitempty = id
tenant-name,omitempty = 
trust-id,omitempty = 
domain-id,omitempty = 
domain-name,omitempty = domain
region,omitempty = eu-de
ca-file,omitempty = 

[LoadBalancer,omitempty]
lb-version,omitempty = 
use-octavia,omitempty = false
subnet-id,omitempty = 4ba20e2c-bb89-468f-81c6-2c67a4bac419
floating-network-id,omitempty = 0a2228f2-7f8a-45f1-8e09-9039e1d09975
lb-method,omitempty = 
lb-provider,omitempty = 
create-monitor,omitempty = false
monitor-delay,omitempty = 0
monitor-timeout,omitempty = 0
monitor-max-retries,omitempty = 0
manage-security-groups,omitempty = false

[BlockStorage,omitempty]
bs-version,omitempty = 
trust-device-path,omitempty = false
ignore-volume-az,omitempty = false

[Route,omitempty]
router-id,omitempty = 

[Metadata,omitempty]
search-order,omitempty = 
request-timeout,omitempty = 0
```

See [comment](https://github.com/rancher/rke/pull/667#issuecomment-398708511)

**Comment by galal-hussein** [comment]
I tested the issue with the latest RKE and i can't reproduce the problem with RKE, i can only reproduce it with Rancher UI, i checked the API and i can see that the config is normal:
```
"cloudProvider": {
"name": "openstack",
"openstackCloudProvider": {
"blockStorage": {
"bs-version": "",
"ignore-volume-az": false,
"trust-device-path": false,
"type": "/v3/schemas/blockStorageOpenstackOpts"
},
"global": {
"auth-url": "https://example.com/v3",
"ca-file": "",
"domain-id": "",
"domain-name": "domain",
"password": "password",
"region": "eu-de",
"tenant-id": "id",
"tenant-name": "",
"trust-id": "",
"type": "/v3/schemas/globalOpenstackOpts",
"user-id": "",
"username": "user"
},
"loadBalancer": {
...
```

**Comment by alexandrem** [comment]
But API has same behavior, so it must be in Rancher code base and not only UI.

**Comment by loganhz** [comment]
I think UI sends the correct payload to backend. If you view API of the newly created cluster, you can see the data has been populated correctly. It's not an UI issue I guess.

**Comment by deniseschannon** [comment]
@dabeck @alexandrem can you test with v2.0.5-rc2 for me? I don't have an openstack setup, and it might be quicker if you could give it a test. It ended up being a vendor dependency issue which was malforming the openstack template, which is why it worked in RKE CLI and not in Rancher.

**Comment by dabeck** [comment]
@deniseschannon Unfortunately I've got no sandbox system at hand right now. But I think you actually don't need a real OpenStack environment. You just need to create a cluster with the cloud_provider setting and check if the file `/etc/kubernetes/cloud-config` in kubelet-Container is well formatted.

**Comment by alexandrem** [comment]
@deniseschannon It is resolved with v2.0.5-rc2

Thanks!

**Comment by deniseschannon** [comment]
Thanks @alexandrem for testing it out!
